MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -3629,8 +3629,15 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 		id<MTLDevice> sysMTLDevice = MTLCreateSystemDefaultDevice();
 		if (mvkGetRegistryID(sysMTLDevice) == mvkGetRegistryID(mtlDevice)) {
 			_physicalDevice->replaceMTLDevice(sysMTLDevice);
+			mtlDevice = _physicalDevice->getMTLDevice();
 		}
 	}
+    
+#if MVK_XCODE_14_3 && TARGET_OS_OSX
+    [mtlDevice setShouldMaximizeConcurrentCompilation: mvkConfig().shouldMaximizeConcurrentCompilation];
+    MVKLogInfoIf(mvkConfig().debugMode, "maximumConcurrentCompilationTaskCount %lu", mtlDevice.maximumConcurrentCompilationTaskCount);
+#endif
```

Comment:
In addition to the Xcode version test, these calls must be wrapped in a `respondsToSelector:` test as well (as found elsewhere), because the runtime might be an older OS version that doesn't support the calls. 

One test for `shouldMaximizeConcurrentCompilation` should be sufficient since `maximumConcurrentCompilationTaskCount` was introduced at the same time.

Reply:
Thanks, updated.